### DISCUSSION TITLE
Network Protocols for firewall logs?

### DISCUSSION THREAD

**Discussion by arisentow** [Security-Onion-Solutions/securityonion#9026] [Unsupported Versions] [answered]
Should network protocols be parsed/translated for pfSense firewall logs in the firewall dataset?

SO Version: 2.3.181
pfSense: 2.6

**Comment by dougburks** [comment, marked as answer]
Are you referring to the `network.transport` field?

![image](https://user-images.githubusercontent.com/1659467/199221257-c69ef523-c1e6-4ef2-8f45-d606eda23531.png)

**Reply by arisentow** [reply]
Sorry, should have specified... "port to service" i.e.- 53 = DNS, 80 = HTTP, etc.

**Reply by dougburks** [reply]
pfSense firewall logs include port numbers but I don't believe they include the actual protocol so there is no protocol for us to parse out of the log. 

We don't automatically translate port number to protocol because any protocol can run over any port. Therefore, you can't always assume that, for example, port 80 traffic is HTTP.
